src/gsp.py

```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
     return forecasts
 
 
+# define route => /v0/solar/GB/gsp/forecast/{gsp_id}/{only_values} or other filter parameters
+# get latest gsp with only values
+# if getting latest
+#  async def get latest only values
+#    logger.info(f"Get forecasts for gsp id {gsp_id} with {forecast_horizon_minutes=}")
+
+# # return get_latest_forecast_values_for_a_specific_gsp_from_database(
+#     session=session,
+#     gsp_id=gsp_id,
+#     forecast_horizon_minutes=forecast_horizon_minutes,
+# )
+
+
+@router.get("/forecast/{gsp_id}/{only_values}", response_model=Union[Forecast, List[Forecast]])
+async def get_forecasts_for_a_specific_gsp(
+    gsp_id: int,
+    session: Session = Depends(get_session),
+    historic: Optional[bool] = False,
+    only_values: Optional[bool] = False,
+    forecast_horizon_minutes: Optional[int] = None,
+) -> Union[Forecast, List[Forecast]]:
+    """### Get one forecast for a specific GSP
+
+    The return object is a solar forecast with GSP system details.
+
+    The forecast object is returned with expected megawatt generation at a specific GSP
+    for the upcoming 8 hours at every 30-minute interval (targetTime).
+
+    Setting history to TRUE on this route will return readings from yesterday and today
+    for the given GSP.
+
+    Please refer to the __Forecast__ and __ForecastValue__ schemas below for metadata definitions.
+
+    #### Parameters
+    - gsp_id: gsp_id of the desired forecast
+    - historic: boolean => TRUE returns yesterday's forecasts in addition to today's forecast
+    """
+
+    logger.info(
+        f"Get forecasts for gsp id {gsp_id} with {historic=} or {only_values} and {forecast_horizon_minutes=}"
+    )
+
+    if only_values is False:
+        full_forecast = get_forecasts_for_a_specific_gsp_from_database(
+            session=session,
+            gsp_id=gsp_id,
+            historic=historic,
+        )
+
+        full_forecast.normalize()
+
+        return full_forecast
+
+    print("this is working")
+
+    only_values_forecast = get_latest_forecast_values_for_a_specific_gsp_from_database(
```

Comment:
goo to do and 
```
if :

else:
```

Just to make it really clear what is going on

Reply:
i had an else there but then deleted it. i'll put it back in.

Reply:
thanks